File: backend/app/api/v1/schemas/job.py
# JobStatus enum, JobResponse, JobResultResponse

from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List
from enum import Enum


class JobStatus(str, Enum):
    SUBMITTED = "SUBMITTED"
    VALIDATING = "VALIDATING"
    QUEUED = "QUEUED"
    PROCESSING = "PROCESSING"
    COMPLETED = "COMPLETED"
    PARTIAL_SUCCESS = "PARTIAL_SUCCESS"
    FAILED = "FAILED"
    REJECTED = "REJECTED"
    CANCELLED = "CANCELLED"
    DEAD_LETTER = "DEAD_LETTER"


class ErrorEntry(BaseModel):
    error: str
    retriable: bool
    timestamp: datetime


class JobResponse(BaseModel):
    id: str
    request_id: str
    file_id: str
    status: JobStatus
    method: str
    tier: int
    retry_count: int
    max_retries: int
    error_history: List[ErrorEntry] = []
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    processing_time_ms: Optional[int] = None
    result_path: Optional[str] = None
    worker_id: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True


class JobResultMetadata(BaseModel):
    method: str
    tier: str
    processing_time_ms: int
    version: str


class JobResultResponse(BaseModel):
    text: str
    lines: int
    metadata: JobResultMetadata

File: 01_frontend/src/types/job.ts
// Job types

export type JobStatus =
  | 'SUBMITTED'
  | 'VALIDATING'
  | 'QUEUED'
  | 'PROCESSING'
  | 'COMPLETED'
  | 'PARTIAL_SUCCESS'
  | 'FAILED'
  | 'REJECTED'
  | 'CANCELLED'
  | 'DEAD_LETTER'

export interface Job {
  id: string
  request_id: string
  file_id: string
  status: JobStatus
  method: string
  tier: number
  output_format: string
  retry_count: number
  max_retries: number
  error_history: ErrorEntry[]
  started_at: string | null
  completed_at: string | null
  processing_time_ms: number | null
  result_path: string | null
  worker_id: string | null
  created_at: string
}

export interface ErrorEntry {
  error: string
  retriable: boolean
  timestamp: string
}

export interface JobResult {
  text: string
  lines: number
  metadata: JobResultMetadata
}

export interface JobResultMetadata {
  method: string
  tier: string
  processing_time_ms: number
  service_version: string
  engine_name: string | null
}
